src/main/java/levi9/VehicleService/repository/RentRepository.java

```diff
@@ -2,13 +2,21 @@
 
 import levi9.VehicleService.model.Rent;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Lock;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.jpa.repository.QueryHints;
 
+import javax.persistence.LockModeType;
+import javax.persistence.QueryHint;
 import java.time.LocalDateTime;
 import java.util.List;
 
 
 public interface RentRepository extends JpaRepository<Rent, Long> {
     List<Rent> findByVehicleId(Long id);
-
     Rent findByClientIdAndEndTimeEquals(Long clientId, LocalDateTime endTime);
+    @Lock(LockModeType.PESSIMISTIC_WRITE)
```

Comment:
This is maybe too much for this use case, it means that it will lock both reads and writes of rentings. Try to implement some kind of optimistic locking.